src/routes/helpers/PrivateRoute.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { Navigate } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { getUser } from '../../store/selectors';
+import { ROUTES } from '../routes';
+
+function PrivateRoute(props) {
```

Comment:
privateroute should also take component as a parameter and render if needed

Reply:
It's a Private Route implementation in react-router v6.
Example: https://dev.to/iamandrewluca/private-route-in-react-router-v6-lg5
Example: https://www.robinwieruch.de/react-router-private-routes/